ui/app/services/persona.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Service from '@ember/service';
+import { service } from '@ember/service';
+
+// import type FeatureFlagService from 'vault/services/feature-flag';
+import type VersionService from 'vault/services/version';
+import type StoreService from 'vault/services/store';
+
+/**
+ * This service returns a persona which can be used to hide or show various states. Currently being used for Secrets Sync, but designed so that other personas can be added.
+ */
+
+export default class PersonaService extends Service {
+  // @service declare readonly featureFlag: FeatureFlagService;
+  @service declare readonly store: StoreService;
+  @service declare readonly version: VersionService;
+
+  get isManagedNamespaceRoot() {
+    // TODO when we get the all clear to allow HVD to use Secrets Sync we can uncomment this.
+    return false;
+    // return this.featureFlag.managedNamespaceRoot ? true : false;
+  }
+
+  /**
+ * Secret Sync persona options are: ['SHOW_ENTERPRISE_CTA, SHOW_PREMIUM_CTA, SHOW_ACTIVATION_CTA and SHOW_SECRETS_SYNC]. The persona return is based on the following criteria:
+  * Community/OSS cluster:
+    1. - Secrets Sync is not available, return SHOW_ENTERPRISE_CTA. This will never show for managed clusters because they are always Enterprise.
+  * Managed cluster:
+    2. - Secrets Sync is activated, return SHOW_SECRETS_SYNC.
+    3. - Secrets Sync is not activated, return SHOW_ACTIVATION_CTA.
+  * On-prem Enterprise cluster:
+    4. - Secrets Sync is on the license and not activated, return SHOW_ACTIVATION_CTA.
+    5. - Secrets Sync is on the license and activated, return SHOW_SECRETS_SYNC.
+    6. - Secrets Sync is not on the license, return SHOW_PREMIUM_CTA.
```

Comment:
i'm curious about how you landed on the naming of these strings here -- wanna fill me in? i was initially imagining that `secretsSyncPersona` would return descriptors of the "type" of user we're assessing vs. the behavior of the UI. my thinking was that we (as developers) usually refer to the type of user when we're trying to figure out whether a user should/shouldn't see something in the UI. for example if it were set up like this:

```typescript
enum SyncPersonas {
  OSS = 'OSS',
  ENT_NOT_ON_LICENSE = 'ENT_NOT_ON_LICENSE',
  ENT_NOT_ACTIVATED = 'ENT_NOT_ACTIVATED',
  ENT_ACTIVATED = 'ENT_ACTIVATED'
}

export default class PersonaService extends Service {
  // ...
  get secretSyncPersona() {
    if (!this.version.isEnterprise) return SyncPersonas.OSS;

    const isHvd = this.featureFlag.managedNamespaceRoot;
    const isActivated = this.version.secretsSyncIsActivated;

    if (isHvd) {
      return isActivated 
        ? SyncPersonas.ENT_ACTIVATED 
        : SyncPersonas.ENT_NOT_ACTIVATED;
    }

    if (this.version.hasSecretsSync) {
      return isActivated 
        ? SyncPersona.ENT_ACTIVATED 
        : SyncPersona.ENT_NOT_ACTIVATED;
    }

    return SyncPersona.ENT_NOT_ON_LICENSE;
  }
}
```

or if you wanted to be really fancy ...
```typescript
export default class PersonaService extends Service {
  // ...
  get secretSyncPersona() {
    if (!this.version.isEnterprise) return SyncPersonas.OSS;

    const isHvd = this.featureFlag.managedNamespaceRoot;
    const syncIsActivated = this.version.secretsSyncIsActivated;
    const hasSyncOnLicense = this.version.hasSecretsSync;

    if ((isHvd || hasSyncOnLicense) && syncIsActivated) {
      return SyncPersonas.ENT_ACTIVATED;
    }

    if ((isHvd || hasSyncOnLicense) && !syncIsActivated) {
      return SyncPersonas.ENT_NOT_ACTIVATED;
    }
 
    return SyncPersona.ENT_NOT_ON_LICENSE;
  }
}
```

the benefit of setting up the result of `get secretsSyncPersona` this was is that it means our use of it within other components/routes will look like this:

```hbs
{{#if (eq @secretsSyncPersona "ENT_ACTIVATED")}}
      <Hds::Button @text="Create first destination" @route="secrets.destinations.create" data-test-cta-button />
{{/if}
      
  {{!-- or --}}

  {{#if (or (eq @secretsSyncPersona "ENT_ACTIVATED") (eq @secretsSyncPersona "ENT_NOT_ACTIVATED"))}}
    <p>
      Sync secrets to platforms and tools across your stack to get secrets when and where you need them.
      <Hds::Link::Standalone
        @icon="learn-link"
        @text="Secrets Sync tutorial"
        @href={{doc-link "/vault/tutorials/enterprise/secrets-sync"}}
        data-test-cta-doc-link
      />
    </p>
  {{else}}
    <p>
      This premium enterprise feature allows you to sync secrets to platforms and tools across your stack to get secrets when and where you need them.
    </p>
    {{/if}}
```

the reason i like the conditionals set up this way is it matches how we talk about whether to show/hide things in conversation. i've noticed we tend to say "we show the activation modal if the user is enterprise and has it on their license but hasn't activated it" or "if the user is OSS we show the premium badge". we don't say "we show the premium badge if we should show the premium badge" 😛 

keeping the persona strings more tightly coupled to the way we refer to them in conversation means we don't have to go through as many mental hoops to try to connect code with our mental models. does that make sense? hope i'm explaining myself well. 😅

anyway, i _do_ recall you picked these strings for a reason, so i'd love to hear yours thoughts on it!

Reply:
Great feedback! 

three thoughts:
1. **persona naming:** In my 🧠 I was thinking about the developer later down the line who was reading the logic expressions on the `landing-cta` component. My thought process was that reading something like `SHOW_SECRETS_SYNC` vs `ENT_ACTIVATED` would help them "read" the template. In a way it takes the step of "why does ent_activated mean show this thing?" out of it.

2.  To your point about how we **talk about templates**, I think changing how we talk about them was my implicit goal. For me, during the 1.16.3 work, the confusing part was trying to keep think through what something like ENT_ACTIVATED meant in context to what view they should be exposed to. However, I do like your phrasing about this point, so let's bring this up as a team discussion?

3. Lastly, I think if we defer to naming that you suggested, we are hinting that all this should go in the `version` service, which more or less returns getters about what is on a user's license, ex: isCommunity, hasDRReplication. And maybe that discovery means we don't really need this persona service and instead should just add more getters to the version service....
🤔

Reply:
oh, i definitely see where you're coming from. 
> My thought process was that reading something like SHOW_SECRETS_SYNC vs ENT_ACTIVATED would help them "read" the template. In a way it takes the step of "why does ent_activated mean show this thing?" out of it

great point! i'm on board with this naming. if we get confused about what `SHOW_SECRETS_SYNC` means it means we have a central spot to find that out, so that addresses my concern 👍 

regarding 3 & 4 -- i've been wondering what the delineation between the `version` service, `feature-flag` service, and `activation-flags` endpoint was, so that's helpful context to have. i'd be wary of adding more getters to the `version` service if its intention is to return things strictly related to `license/features` though. and while an `activation-features` service may be helpful, my understanding is regardless we still need a way to check for things that take into account all 3 sources of truth: the `license/features`, `activated-features`, and `internal/ui/feature-flags`. and that check needs to be reusable, right? (i may be wrong!)

i'm assuming that we'd want to check for the same things when determining whether to show the badges in the nav bar, the CTA on the overview page, or the sync tab on the client counts page. but we may not need to. it may help if we start implementing those checks and notice whether we're making the same checks in multiple places, and if so, then consolidate them into a service like what we have here. what do you think? that can be something we pair on tomorrow if you'd like 🙂

Reply:
>it may help if we start implementing those checks and notice whether we're making the same checks in multiple places, and if so, then consolidate them into a service like what we have here.

I think that's a great idea. And re: the delineation of the services, that's something I observed but I'm pretty sure was never the explicit intention. I'll bring this up during standup to see if as a task the team wants to move any activation-features requests out of the version service and into their own service to keep that delineation true.

Reply:
Correction on 4 above, the version service is currently scoped to two endpoints, not one. It's the seal-status and license/features endpoints.